sql/scan.go

```diff
@@ -363,6 +371,66 @@ func (n *scanNode) initTargets(targets parser.SelectExprs) error {
 	return nil
 }
 
+var (
+	errLimitNoVariables  = errors.New("argument of LIMIT must not contain variables")
+	errOffsetNoVariables = errors.New("argument of OFFSET must not contain variables")
+)
+
+func (n *scanNode) initLimit(limitNode *parser.Limit) error {
+	if limitNode == nil {
+		n.limit, n.offset = parser.DNull, parser.DNull
+		return nil
+	}
+
+	if limitNode.Count == nil {
+		n.limit = parser.DNull
+	} else {
+		if parser.ContainsVars(limitNode.Count) {
+			n.err = errLimitNoVariables
+			return n.err
+		}
+
+		limitNode.Count, n.err = parser.NormalizeExpr(limitNode.Count)
+		if n.err != nil {
+			return n.err
+		}
+		n.limit, n.err = parser.EvalExpr(limitNode.Count)
+		if n.err != nil {
+			return n.err
+		}
+
+		if _, ok := n.limit.(parser.DInt); !(ok || n.limit == parser.DNull) {
+			n.err = fmt.Errorf("argument of OFFSET must be type %s, not type %s", parser.DummyInt.Type(), n.limit.Type())
```

Comment:
`s/OFFSET/LIMIT/g`, though see my comment below about unifying the handling of `Count` and `Offset`.